### PR TITLE
[Feature] Add TruncNorm initializer for BERT

### PR DESCRIPTION
**Pull request by Ishitori**
## Description ##
BERT requires TruncNorm initializer. This code uses scipy to draw samples from distribution as a quick solution. If it would be too slow for bigger sample sizes, will have to rewrite it.

### Essentials ###
- [X] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [X] Changes are complete (i.e. I finished coding on this PR)
- [X] All changes have test coverage
- [X] Code is well-documented

### CODE REVIEW

File: src/gluonnlp/initializer/initializer.py [hunk 1]
```diff
@@ -63,3 +62,17 @@ def _init_weight(self, name, arr):
         """Abstract method to Initialize weight."""
         arr[:int(arr.shape[0] / 2)] = self.nonlinear_transform_bias
         arr[int(arr.shape[0] / 2):] = self.transform_gate_bias
+
+
+@mxnet.initializer.register
+class TruncNorm(Initializer):
+    r"""Initialize the weight by drawing sample from truncated normal distribution with
+    provided mean and standard deviation. Values whose magnitude is more than 2 standard deviations
+    from the mean are dropped and re-picked.."""
```
[the remark this answers is not in the document]
Comment (Ishitori) [reply src/gluonnlp/initializer/initializer.py]: Done

**Comment by eric-haibin-lin** [comment]
Could you fix the lint failure?

**Comment by mli** [comment]
Job PR-548/4 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-548/4/index.html

File: src/gluonnlp/initializer/initializer.py [hunk 2]
```diff
@@ -63,3 +62,31 @@ def _init_weight(self, name, arr):
...
+
+    stdev : float, default 0.01
+        Standard deviation of the underlying normal distribution
+
+    **kwargs : dict
+        Additional parameters for base Initializer.
+    """
+    def __init__(self, mean=0, stdev=0.01, **kwargs):
+        super(TruncNorm, self).__init__(**kwargs)
+        from scipy.stats import truncnorm
```
Comment (szha) [inline src/gluonnlp/initializer/initializer.py]: catch import error

**Comment by mli** [comment]
Job PR-548/6 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-548/6/index.html

**Pull request merged**